File: app/sdks/python/appwrite/services/storage.py
from ..service import Service


class Storage(Service):

    def __init__(self, client):
        super(Storage, self).__init__(client)

    def list_files(self, search='', limit=25, offset=0, order_type='ASC'):
        """List Files"""

        params = {}
        path = '/storage/files'
        params['search'] = search
        params['limit'] = limit
        params['offset'] = offset
        params['orderType'] = order_type

        return self.client.call('get', path, {
            'content-type': 'application/json',
        }, params)

    def create_file(self, files, read, write, folder_id=''):
        """Create File"""

        params = {}
        path = '/storage/files'
        params['files'] = files
        params['read'] = read
        params['write'] = write
        params['folderId'] = folder_id

        return self.client.call('post', path, {
            'content-type': 'multipart/form-data',
        }, params)

    def get_file(self, file_id):
        """Get File"""

        params = {}
        path = '/storage/files/{fileId}'
        path.replace('{fileId}', file_id)                

        return self.client.call('get', path, {
            'content-type': 'application/json',
        }, params)

    def update_file(self, file_id, read, write, folder_id=''):
        """Update File"""

        params = {}
        path = '/storage/files/{fileId}'
        path.replace('{fileId}', file_id)                
        params['read'] = read
        params['write'] = write
        params['folderId'] = folder_id

        return self.client.call('put', path, {
            'content-type': 'application/json',
        }, params)

    def delete_file(self, file_id):
        """Delete File"""

        params = {}
        path = '/storage/files/{fileId}'
        path.replace('{fileId}', file_id)                

        return self.client.call('delete', path, {
            'content-type': 'application/json',
        }, params)

    def get_file_download(self, file_id):
        """Get File for Download"""

        params = {}
        path = '/storage/files/{fileId}/download'
        path.replace('{fileId}', file_id)                

        return self.client.call('get', path, {
            'content-type': 'application/json',
        }, params)

    def get_file_preview(self, file_id, width=0, height=0, quality=100, background='', output=''):
        """Get File Preview"""

        params = {}
        path = '/storage/files/{fileId}/preview'
        path.replace('{fileId}', file_id)                
        params['width'] = width
        params['height'] = height
        params['quality'] = quality
        params['background'] = background
        params['output'] = output

        return self.client.call('get', path, {
            'content-type': 'application/json',
        }, params)

    def get_file_view(self, file_id, as=''):
        """Get File for View"""

        params = {}
        path = '/storage/files/{fileId}/view'
        path.replace('{fileId}', file_id)                
        params['as'] = as

        return self.client.call('get', path, {
            'content-type': 'application/json',
        }, params)

File: app/sdks/0.6.2/server-java/src/main/java/services/Storage.java
package .services;



import okhttp3.Call;
import okhttp3.HttpUrl;
import .Client;
import .enums.OrderType;

import java.io.File;
import java.util.List;
import java.util.HashMap;
import java.util.Map;

import static java.util.Map.entry;

public class Storage extends Service {
    public Storage(Client client){
        super(client);
    }

     /// List Files
     /*
     * Get a list of all the user files. You can use the query params to filter
     * your results. On admin mode, this endpoint will return a list of all of the
     * project files. [Learn more about different API modes](/docs/admin).
     */
    public Call listFiles(String search, int limit, int offset, OrderType orderType) {
        final String path = "/storage/files";

        final Map<String, Object> params = Map.ofEntries(
            entry("search", search),
            entry("limit", limit),
            entry("offset", offset),
            entry("orderType", orderType.name())
        );



        final Map<String, String> headers = Map.ofEntries(
            entry("content-type", "application/json")
        );

        return client.call("GET", path, headers, params);
    }

     /// Create File
     /*
     * Create a new file. The user who creates the file will automatically be
     * assigned to read and write access unless he has passed custom values for
     * read and write arguments.
     */
    public Call createFile(File file, List read, List write) {
        final String path = "/storage/files";

        final Map<String, Object> params = Map.ofEntries(
            entry("file", file),
            entry("read", read),
            entry("write", write)
        );



        final Map<String, String> headers = Map.ofEntries(
            entry("content-type", "multipart/form-data")
        );

        return client.call("POST", path, headers, params);
    }

     /// Get File
     /*
     * Get file by its unique ID. This endpoint response returns a JSON object
     * with the file metadata.
     */
    public Call getFile(String fileId) {
        final String path = "/storage/files/{fileId}".replace("{fileId}", fileId);

        final Map<String, Object> params = Map.ofEntries(
        );



        final Map<String, String> headers = Map.ofEntries(
            entry("content-type", "application/json")
        );

        return client.call("GET", path, headers, params);
    }

     /// Update File
     /*
     * Update file by its unique ID. Only users with write permissions have access
     * to update this resource.
     */
    public Call updateFile(String fileId, List read, List write) {
        final String path = "/storage/files/{fileId}".replace("{fileId}", fileId);

        final Map<String, Object> params = Map.ofEntries(
            entry("read", read),
            entry("write", write)
        );



        final Map<String, String> headers = Map.ofEntries(
            entry("content-type", "application/json")
        );

        return client.call("PUT", path, headers, params);
    }

     /// Delete File
     /*
     * Delete a file by its unique ID. Only users with write permissions have
     * access to delete this resource.
     */
    public Call deleteFile(String fileId) {
        final String path = "/storage/files/{fileId}".replace("{fileId}", fileId);

        final Map<String, Object> params = Map.ofEntries(
        );



        final Map<String, String> headers = Map.ofEntries(
            entry("content-type", "application/json")
        );

        return client.call("DELETE", path, headers, params);
    }

     /// Get File for Download
     /*
     * Get file content by its unique ID. The endpoint response return with a
     * 'Content-Disposition: attachment' header that tells the browser to start
     * downloading the file to user downloads directory.
     */
    public String getFileDownload(String fileId) {
        final String path = "/storage/files/{fileId}/download".replace("{fileId}", fileId);

        final Map<String, Object> params = Map.ofEntries(
            entry("project", client.getConfig().get("project")),
            entry("key", client.getConfig().get("key"))
        );



        HttpUrl.Builder httpBuilder = new HttpUrl.Builder().build().newBuilder(client.getEndPoint() + path);
        params.forEach((k, v) -> httpBuilder.addQueryParameter(k, v.toString()));

        return httpBuilder.build().toString();
    }

     /// Get File Preview
     /*
     * Get a file preview image. Currently, this method supports preview for image
     * files (jpg, png, and gif), other supported formats, like pdf, docs, slides,
     * and spreadsheets, will return the file icon image. You can also pass query
     * string arguments for cutting and resizing your preview image.
     */
    public String getFilePreview(String fileId, int width, int height, int quality, String background, String output) {
        final String path = "/storage/files/{fileId}/preview".replace("{fileId}", fileId);

        final Map<String, Object> params = Map.ofEntries(
            entry("width", width),
            entry("height", height),
            entry("quality", quality),
            entry("background", background),
            entry("output", output),
            entry("project", client.getConfig().get("project")),
            entry("key", client.getConfig().get("key"))
        );



        HttpUrl.Builder httpBuilder = new HttpUrl.Builder().build().newBuilder(client.getEndPoint() + path);
        params.forEach((k, v) -> httpBuilder.addQueryParameter(k, v.toString()));

        return httpBuilder.build().toString();
    }

     /// Get File for View
     /*
     * Get file content by its unique ID. This endpoint is similar to the download
     * method but returns with no  'Content-Disposition: attachment' header.
     */
    public String getFileView(String fileId, String as) {
        final String path = "/storage/files/{fileId}/view".replace("{fileId}", fileId);

        final Map<String, Object> params = Map.ofEntries(
            entry("as", as),
            entry("project", client.getConfig().get("project")),
            entry("key", client.getConfig().get("key"))
        );



        HttpUrl.Builder httpBuilder = new HttpUrl.Builder().build().newBuilder(client.getEndPoint() + path);
        params.forEach((k, v) -> httpBuilder.addQueryParameter(k, v.toString()));

        return httpBuilder.build().toString();
    }
}